src/decode/lzma.rs

```diff
@@ -152,20 +145,9 @@ where
 }
 
 // Initialize decoder with accumulating buffer
-pub fn new_accum<W>(
-    output: lzbuffer::LzAccumBuffer<W>,
-    lc: u32,
-    lp: u32,
-    pb: u32,
-    unpacked_size: Option<u64>,
-) -> DecoderState<W, lzbuffer::LzAccumBuffer<W>>
-where
-    W: io::Write,
-{
+pub fn new_accum(lc: u32, lp: u32, pb: u32, unpacked_size: Option<u64>) -> DecoderState {
     DecoderState {
```

Comment:
I think this was missed before, but can we add some assertions on `lc`, `lp` and `pb` to prevent invalid values? I.e. `lc < 8`, `lp < 4` and `pb < 4`?

Reply:
https://github.com/gendx/lzma-rs/pull/73/files#diff-768757151baad4fb0605aaeee5210dc343977125d64711ebcf5a0e755b2f23c0R150